src/utils/crypto.js

```diff
@@ -39,24 +39,34 @@ function hmacJsonForPskCondition (obj, sharedSecret) {
 
 // turn object into encrypted buffer
 function aesEncryptObject (obj, sharedSecret) {
+  return aesEncryptBuffer(Buffer.from(JSON.stringify(obj), 'utf8'))
+}
+
+// turn buffer into encrypted buffer
+function aesEncryptBuffer (buffer, sharedSecret) {
   const pskEncryptionKey = hmac(sharedSecret, PSK_ENCRYPTION_STRING)
   const cipher = crypto.createCipher('aes-256-ctr', pskEncryptionKey)
 
   return Buffer.concat([
-    cipher.update(Buffer.from(JSON.stringify(obj), 'utf8')),
+    cipher.update(buffer),
     cipher.final()
   ])
 }
 
-// turn base64-encoded encrypted text into parsed object
-function aesDecryptObject (encrypted, sharedSecret) {
+// turn buffer into decrypted buffer
+function aesDecryptBuffer (encrypted, sharedSecret) {
   const pskEncryptionKey = hmac(sharedSecret, PSK_ENCRYPTION_STRING)
   const decipher = crypto.createDecipher('aes-256-ctr', pskEncryptionKey)
 
-  const decoded = Buffer.concat([
-    decipher.update(Buffer.from(encrypted, 'base64')),
+  return Buffer.concat([
+    decipher.update(encrypted),
```

Comment:
https://nodejs.org/api/crypto.html#crypto_class_decipher that's not how the decipher class works, according to the docs